include/lemon/matrix.hpp

```diff
@@ -0,0 +1,280 @@
+#ifndef LEMON_MATRIX_HPP
+#define LEMON_MATRIX_HPP
+
+#include <chemfiles/types.hpp>
+
+#include <array>
+#include <cmath>
+#include <complex>
+#include <vector>
+#include <algorithm>
+
+namespace lemon {
+
+using Vector3D = chemfiles::Vector3D;
+using Matrix3D = chemfiles::Matrix3D;
+using Coordinates = std::vector<Vector3D>;
+
+using chemfiles::cross;
+using chemfiles::dot;
+
+inline double trace(const Matrix3D& matrix) {
+    return matrix[0][0] + matrix[1][1] + matrix[2][2];
+}
+
+template <typename Container = Coordinates>
+inline Matrix3D covariant(const Container& a, const Container& b) {
+
+    Matrix3D result = Matrix3D::zero();
+
+    for (size_t i = 0; i < 3; ++i) {
+        for (size_t j = 0; j < 3; ++j) { // transpose of b, not b
+            for (size_t k = 0; k < a.size(); ++k) {    // We are stuck in 3 dimensions
+                result[i][j] += a[k][i] * b[k][j]; // transpose of b, not b
+            }
+        }
+    }
+
+    return result;
+}
+
+template <typename Container = Coordinates>
+inline double rmsd(const Container& a, const Container& b) {
+    double result = 0;
+    for (size_t i = 0; i < a.size(); ++i) {
```

Comment:
this will fail if called with b.size()<a.size(). Maybe check/assert that sizes are equal?

Reply:
Yes, this a good idea!